pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -791,3 +825,409 @@ func TestGetTPMapMatchingIncomingAffinityAntiAffinity(t *testing.T) {
+				{key: "node", value: "node-a"}: {1},
+				{key: "node", value: "node-b"}: {},
+				{key: "node", value: "node-y"}: {4, 6},
+			},
+			want: &topologyPairsPodSpreadMap{
+				minMatchMap: map[string]int32{"zone": 1, "node": 0},
+				topologyPairsMaps: &topologyPairsMaps{
+					podToTopologyPairs: map[string]topologyPairSet{
+						"p-a2_": newPairSet("zone", "zone1", "node", "node-a"),
+						"p-y2_": newPairSet("zone", "zone2", "node", "node-y"),
+						"p-y4_": newPairSet("zone", "zone2", "node", "node-y"),
+					},
+				},
+			},
+		},
+		{
+			name: "two spreadConstraints, and with podAffinity",
+			pod: u.MakePod().Name("p").Label("foo", "").
+				NodeAffinityNotIn("node", []string{"node-x"}). // exclude node-x
+				SpreadConstraint(1, "zone", hardSpread, u.MakeLabelSelector().Exists("foo").Obj()).
+				SpreadConstraint(1, "node", hardSpread, u.MakeLabelSelector().Exists("foo").Obj()).
+				Obj(),
+			nodes: []*v1.Node{
+				u.MakeNode().Name("node-a").Label("zone", "zone1").Label("node", "node-a").Obj(),
+				u.MakeNode().Name("node-b").Label("zone", "zone1").Label("node", "node-b").Obj(),
+				u.MakeNode().Name("node-x").Label("zone", "zone2").Label("node", "node-x").Obj(),
+				u.MakeNode().Name("node-y").Label("zone", "zone2").Label("node", "node-y").Obj(),
+			},
+			existingPods: []*v1.Pod{
+				u.MakePod().Name("p-a1").Node("node-a").Label("foo", "").Obj(),
+				u.MakePod().Name("p-a2").Node("node-a").Label("foo", "").Obj(),
+				u.MakePod().Name("p-b1").Node("node-b").Label("foo", "").Obj(),
+				u.MakePod().Name("p-y1").Node("node-y").Label("foo", "").Obj(),
+				u.MakePod().Name("p-y2").Node("node-y").Label("foo", "").Obj(),
+				u.MakePod().Name("p-y3").Node("node-y").Label("foo", "").Obj(),
+				u.MakePod().Name("p-y4").Node("node-y").Label("foo", "").Obj(),
+			},
+			injectPodPointers: map[topologyPair][]int{
+				{key: "zone", value: "zone1"}:  {0, 1, 2},
+				{key: "zone", value: "zone2"}:  {3, 4, 5, 6},
+				{key: "node", value: "node-a"}: {0, 1},
+				{key: "node", value: "node-b"}: {2},
+				{key: "node", value: "node-y"}: {3, 4, 5, 6},
+			},
+			want: &topologyPairsPodSpreadMap{
+				minMatchMap: map[string]int32{"zone": 3, "node": 1},
+				topologyPairsMaps: &topologyPairsMaps{
+					podToTopologyPairs: map[string]topologyPairSet{
+						"p-a1_": newPairSet("zone", "zone1", "node", "node-a"),
+						"p-a2_": newPairSet("zone", "zone1", "node", "node-a"),
+						"p-b1_": newPairSet("zone", "zone1", "node", "node-b"),
+						"p-y1_": newPairSet("zone", "zone2", "node", "node-y"),
+						"p-y2_": newPairSet("zone", "zone2", "node", "node-y"),
+						"p-y3_": newPairSet("zone", "zone2", "node", "node-y"),
+						"p-y4_": newPairSet("zone", "zone2", "node", "node-y"),
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.want.topologyPairToPods = make(map[topologyPair]podSet)
+			for pair, indice := range tt.injectPodPointers {
```

Comment:
s/indice/indices

Reply:
Let's just use `indexes` to avoid confusion.